pkg/strategy/autobuy/strategy.go

```diff
@@ -0,0 +1,146 @@
+package autobuy
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/c9s/bbgo/pkg/bbgo"
+	"github.com/c9s/bbgo/pkg/fixedpoint"
+	indicatorv2 "github.com/c9s/bbgo/pkg/indicator/v2"
+	"github.com/c9s/bbgo/pkg/strategy/common"
+	"github.com/c9s/bbgo/pkg/types"
+	"github.com/robfig/cron/v3"
+	"github.com/sirupsen/logrus"
+)
+
+const ID = "autobuy"
+
+var log = logrus.WithField("strategy", ID)
+
+func init() {
+	bbgo.RegisterStrategy(ID, &Strategy{})
+}
+
+type Strategy struct {
+	*common.Strategy
+
+	Environment *bbgo.Environment
+	Market      types.Market
+
+	Symbol           string                  `json:"symbol"`
+	CronExpression   string                  `json:"cronExpression"`
+	Threshold        fixedpoint.Value        `json:"threshold"`
+	BollingerSetting *types.BollingerSetting `json:"bollingerSetting"`
+	DryRun           bool                    `json:"dryRun"`
+
+	bbgo.QuantityOrAmount
+
+	boll *indicatorv2.BOLLStream
+	cron *cron.Cron
+}
+
+func (s *Strategy) Initialize() error {
+	if s.Strategy == nil {
+		s.Strategy = &common.Strategy{}
+	}
+	return nil
+}
+
+func (s *Strategy) ID() string {
+	return ID
+}
+
+func (s *Strategy) InstanceID() string {
+	return fmt.Sprintf("%s:%s", ID, s.Symbol)
+}
+
+func (s *Strategy) Validate() error {
+	if err := s.QuantityOrAmount.Validate(); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (s *Strategy) Subscribe(session *bbgo.ExchangeSession) {
+	session.Subscribe(types.KLineChannel, s.Symbol, types.SubscribeOptions{Interval: s.BollingerSetting.Interval})
+}
+
+func (s *Strategy) Run(ctx context.Context, _ bbgo.OrderExecutor, session *bbgo.ExchangeSession) error {
+	s.Strategy.Initialize(ctx, s.Environment, session, s.Market, ID, s.InstanceID())
+
+	s.boll = session.Indicators(s.Symbol).BOLL(s.BollingerSetting.IntervalWindow, s.BollingerSetting.BandWidth)
+
+	session.UserDataStream.OnBalanceUpdate(func(balances types.BalanceMap) {
+		s.autobuy(ctx)
+	})
+
+	// the shutdown handler, you can cancel all orders
+	bbgo.OnShutdown(ctx, func(ctx context.Context, wg *sync.WaitGroup) {
+		defer wg.Done()
+		s.cancelOrders(ctx)
+	})
+
+	s.cron = cron.New()
+	s.cron.AddFunc(s.CronExpression, func() {
+		s.autobuy(ctx)
+	})
+	s.cron.Start()
+
+	return nil
+}
+
+func (s *Strategy) cancelOrders(ctx context.Context) {
+	if err := s.OrderExecutor.GracefulCancel(ctx); err != nil {
+		log.WithError(err).Errorf("failed to cancel orders")
+	}
+}
+
+func (s *Strategy) autobuy(ctx context.Context) {
+	s.cancelOrders(ctx)
+
+	balance, ok := s.Session.GetAccount().Balance(s.Market.BaseCurrency)
+	if !ok {
+		log.Errorf("%s balance not found", s.Market.BaseCurrency)
+		return
+	}
+	log.Infof("balance: %s", balance.String())
+
+	ticker, err := s.Session.Exchange.QueryTicker(ctx, s.Symbol)
+	if err != nil {
+		log.WithError(err).Errorf("failed to query ticker")
+		return
+	}
+
+	price := ticker.Buy
```

Comment:
it could also be "ticker.Sell - one tick" if we want to be filled faster?

Reply:
this could be an option

Reply:
How about
```
	var price fixedpoint.Value
	switch s.PriceType {
	case PriceTypeLast:
		price = ticker.Last
	case PriceTypeBuy:
		price = ticker.Buy
	case PriceTypeSell:
		price = ticker.Sell
	case PriceTypeMid:
		price = ticker.Buy.Add(ticker.Sell).Div(two)
	default:
		price = ticker.Buy
	}
```